scripts/a11y_audit.py

```
@@ -0,0 +1,1153 @@
+        self.snippet = snippet.strip()[:160]
+        self.detail = detail
+
+    @property
+    def severity(self) -> str:
+        return RULES[self.rule_id].severity
+
+    def fingerprint(self) -> str:
+        """Stable across line moves and reindentation, unlike a line number."""
+        normalized = re.sub(r"\s+", "", self.snippet)
+        payload = "{}|{}|{}".format(self.rule_id, self.path, normalized)
+        return hashlib.sha1(payload.encode("utf-8")).hexdigest()[:16]
```

Ruff (0.16.2)

[error] Probable use of insecure hash functions in `hashlib`: `sha1`

(S324)